Add Creality Board 452 Support

InsanityAutomation opened the pull request:
Following work by @Sebazzz on the Creality CR6 printer, this breaks out the motherboard support to give a path to support in chunks, minimizing rebase work and potential conflicts as LCD code evolves.

Probes with explicit enable pin coming up next followed by ExtUI is homing extensions.

InsanityAutomation committed:
Add Creality Board 452 Support

sjasonsmith commented on Marlin/src/HAL/STM32F1/HAL.h:
```diff
@@ -109,6 +109,8 @@
     #error "LCD_SERIAL_PORT must be -1 or from 1 to 3. Please update your configuration."
   #endif
+
+  #define SERIAL_GET_TX_BUFFER_FREE LCD_SERIAL.availableForWrite
```
Is this change necessary in this commmit? It seems unrelated to me.

InsanityAutomation replied:
This could go in either commit really but will be beneficial for any serial display on this board so i dropped it here.

sjasonsmith commented on Marlin/src/pins/stm32f1/pins_CREALITY_V452.h:
```diff
@@ -0,0 +1,116 @@
+//
+// EEPROM
+//
+#if NO_EEPROM_SELECTED
+  #define IIC_BL24CXX_EEPROM                      // EEPROM on I2C-0 used only for display settings
+  #if ENABLED(IIC_BL24CXX_EEPROM)
+    #define IIC_EEPROM_SDA                  PA11
+    #define IIC_EEPROM_SCL                  PA12
+    #define MARLIN_EEPROM_SIZE             0x800  // 2Kb (24C16)
+  #else
+    #define SDCARD_EEPROM_EMULATION               // SD EEPROM until all EEPROM is BL24CXX
+    #define MARLIN_EEPROM_SIZE             0x800  // 2Kb
+  #endif
+#endif
```
I just reworked these in pins_Creaity_V4, so that people could choose to use SD EEPROM if they want. This is what I ended up with:

```
//
// EEPROM
//
#if NO_EEPROM_SELECTED
  #define IIC_BL24CXX_EEPROM                      // EEPROM on I2C-0
  //#define SDCARD_EEPROM_EMULATION
#endif

#if ENABLED(IIC_BL24CXX_EEPROM)
  #define IIC_EEPROM_SDA                  PA11
  #define IIC_EEPROM_SCL                  PA12
  #define MARLIN_EEPROM_SIZE             0x800  // 2Kb (24C16)
#elif ENABLED(SDCARD_EEPROM_EMULATION)
  #define MARLIN_EEPROM_SIZE             0x800  // 2Kb
#endif
```

InsanityAutomation replied:
Ill take a closer look if this lines up here as well when the sensor stuff is done.

sjasonsmith replied:
I went ahead and changed this. There is no functional change, other than allowing SDCARD_EEPROM_EMULATION from a config file to properly set MARLIN_EEPROM_SIZE.

sjasonsmith commented on Marlin/src/pins/stm32f1/pins_CREALITY_V452.h:
```diff
@@ -0,0 +1,116 @@
+//
+// Limit Switches
+//
+#define X_MIN_PIN           PC4
+// #define X_MAX_PIN          PA7
```
Does this board potentially have separate Min/Max endstops?

It seems like these should probably just define STOP pins.

InsanityAutomation replied:
I havnt pulled the cover off yet so ill let you know :)

sjasonsmith commented on Marlin/src/pins/stm32f1/pins_CREALITY_V452.h:
```diff
@@ -0,0 +1,116 @@
+// #define X_MAX_PIN          PA7
+#define Y_MIN_PIN           PC5
+#define Z_MIN_PIN           PA4
+#define COM_PIN             PA5
```
What is this? Does this have something to do with the probe?

sjasonsmith replied:
This specifically refers to `COM_PIN`. I'm just not sure what it is.

InsanityAutomation replied:
Yes, ill be renaming it to a TARE pin as I generalize that. When the next PR is finished ill come back and modify this.

sjasonsmith replied:
This is in my push, based on our chat.

sjasonsmith commented on Marlin/src/pins/stm32f1/pins_CREALITY_V452.h:
```diff
@@ -0,0 +1,116 @@
+#define PROBE_ENABLE_PIN    PC6 // Optoswitch to Enable Z Probe
+
+
+#define TEMP_TIMER_CHAN 4 // Channel of the timer to use for compare and interrupts
```
I doubt this is actually needed.

sjasonsmith replied:
Referring to `TEMP_TIMER_CHAN`

InsanityAutomation replied:
Maybe, I would need to follow it back a bit but for the moment all known references have it so I didnt touch it till I had time to investigate.

sjasonsmith replied:
It looks like it is already in the PIO environment, so I removed it here.
It probably isn't _actually_ needed anywhere, but removing it here should not matter.

sjasonsmith committed:
Reorder board numbers

sjasonsmith committed:
Update according to latest pins changes for v4 board
- Define no native USB
- Update EEPRMO definition
- Remove timer channel, which is in PIO file

sjasonsmith committed:
Update checks

sjasonsmith committed:
Copyright year, alignment
